### PR TITLE
[docs] Contributing and releasing doc updates

### PR DESCRIPTION
**Pull request by CodeBlanch**
## Changes

* Update `RELEASING.md` for new "request release" issue added in #2080
* Update `CONTRIBUTING.md` to link to `RELEASING.md`
* Misc updates to `CONTRIBUTING.md` for things that have changed/evolved since it was last visited

## Merge requirement checklist

* [X] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -133,48 +160,42 @@ A PR is considered to be **ready to merge** when:
...
-```text
-## 1.2.0-beta.2
-
-Released 2022-Jun-21
-```
+## Release process
 
-* Tag the maintainers of this repository
-(@open-telemetry/dotnet-contrib-maintainers) who can release the package.
+For details about the relaese process and information about how to request a
```
Comment (cijothomas) [inline CONTRIBUTING.md:178]: Can we link to the specific header within releasing.md file, as that doc has lot of maintainers only content too.

Comment (cijothomas) [inline CONTRIBUTING.md:178]:
```suggestion
For details about the release process and information about how to request a
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -25,19 +34,19 @@ resource for your first contribution, join our Slack channel and find a buddy!
...
 
-You can contribute to this project from a Windows, macOS or Linux machine.
+You can contribute to this project from a Windows, macOS, or Linux machine.
 
 On all platforms, the minimum requirements are:
 
-* Git client and command line tools.
-* .NET 8.0+
+* Git client and command line tools
+* [.NET SDK (latest stable version)](https://dotnet.microsoft.com/download)
```
Comment (cijothomas) [inline CONTRIBUTING.md:47]: is there a need to install latest preview when nearing the .NET release date?

**Review by cijothomas — APPROVED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (CodeBlanch) [reply CONTRIBUTING.md:47]: I don't think in this repo we should really ever be using a pre-release SDK. In main repo that is more valid for sure.

**Pull request merged**